src/main/java/org/broadinstitute/hellbender/utils/GATKProtectedVariantContextUtils.java

```diff
@@ -93,6 +93,48 @@
         return attributeValueToIntArray(genotype.getExtendedAttribute(key), key, defaultValue, missingValue);
     }
 
+
+    /**
+     * Get Long attribute from a variant context.
+     *
+     * @param variantContext the target variant-context.
+     * @param attribute the name of the attribute containing the Long value.
+     * @return never {@code null}.
+     * @throws IllegalArgumentException if {@code variantContext} is {@code null} or {@code key} is {@code null}.
+     */
+    public static Long getAttributeAsLong(final VariantContext variantContext, final String attribute, final Long defaultValue) {
+        Utils.nonNull(variantContext);
+        Utils.nonNull(attribute);
+        Object x = variantContext.getAttribute(attribute);
+        if ( x == null || x == VCFConstants.MISSING_VALUE_v4 ) return defaultValue;
+        if ( x instanceof Number ) return ((Number) x).longValue();
+        return Long.valueOf((String)x); // throws an exception if this isn't a string
+    }
+
+    /**
+     * Composes the Long List from a variant context.
+     *
+     * @param variantContext the target variant-context.
+     * @param attribute the name of the attribute containing the Long list.
+     * @return never {@code null}.
+     * @throws IllegalArgumentException if {@code variantContext} is {@code null} or {@code key} is {@code null}.
+     */
+    public static List<Long> getAttributeAsLongList(final VariantContext variantContext, final String attribute, final Long defaultValue) {
+        Utils.nonNull(variantContext);
+        Utils.nonNull(attribute);
+        return variantContext.getAttributeAsList(attribute).stream().map(
+                x -> {
+                    if (x == null || x == VCFConstants.MISSING_VALUE_v4) {
```

Comment:
There was a bug reported in htsjdk for code that's (nearly) identical identical to this in `CommonInfo.getAttributeAsIntList()`, https://github.com/samtools/htsjdk/issues/1228

It would be nice if there was a way to avoid the duplicate code so we don't have to fix the same bug everywhere.

Reply:
Nice catch noticing this.

Reply:
I'm willing to fix things in htsjdk if that's the decision that people reach. In the meantime I corrected the equality test for the new getter functions.

Reply:
Given that htsjdk 3.0 is going to have a huge type overhaul, I didn't want to agonize over adding a new method to htsjdk.  @lbergelson thoughts?